web/js/codeworld_shared.js

```diff
@@ -1559,78 +1559,84 @@ function preFormatMessage(msg) {
 }
 
 function printMessage(type, message) {
-  const outputDiv = document.getElementById('message');
+  const $outputBlock = $('#message');
 
-  let box = outputDiv.lastChild;
-  let messageContent;
-  if (box && type === 'log' && box.classList.contains('log')) {
-    box.rawMessage += message;
-    messageContent = box.lastChild;
-  } else {
-    box = document.createElement('div');
-    box.classList.add('message-box');
-    box.classList.add(type);
+  let $box = $outputBlock.find('.message-box');
+  let $messageContent = $outputBlock.find('.message-content');
 
-    const messageGutter = document.createElement('div');
-    messageGutter.classList.add('message-gutter');
+  if (!$box.length && !$messageContent.length) {
+    $box = $('<div>');
+    $box.addClass(`message-box ${type}`);
 
-    messageContent = document.createElement('div');
-    messageContent.classList.add('message-content');
+    const $messageGutter = $('<div>');
+    $messageGutter.addClass('message-gutter');
 
-    box.appendChild(messageGutter);
-    box.appendChild(messageContent);
+    $messageContent = $('<div>');
+    $messageContent.addClass('message-content');
 
-    box.rawMessage = message;
+    $box.append($messageGutter, $messageContent);
+    $outputBlock.append($box);
   }
 
-  const formatted = preFormatMessage(box.rawMessage);
-  const lines = formatted.trim().split('\n');
+  const formattedMessage = preFormatMessage(message);
+  const lines = formattedMessage.trim().split('\n');
 
-  messageContent.innerHTML = '';
+  let $details = $messageContent.find('details');
+  let $summary = $messageContent.find('summary');
+  let $firstLine;
 
-  let firstLine;
   if (lines.length < 2) {
```

Comment:
I'm not sure this will correctly handle the transition from a single-line message to a multi-line message, when new content is appended.